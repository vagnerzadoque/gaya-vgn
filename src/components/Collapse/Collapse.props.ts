export type CollapseProps = {
  /**
   * Override or extend the styles applied to the component
   */
  className?: string
  /**
   * Controls when the content should be show or hide
   * @default false
   */
  open?: boolean
  /**
   * Optional ID for testing
   */
  testID?: string

  /**
   * The content of collapse
   */
  children: React.ReactNode
}
