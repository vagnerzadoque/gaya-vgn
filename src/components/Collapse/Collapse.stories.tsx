import React, { useState } from 'react'
import type { Meta, StoryObj } from '@storybook/react'
import { useTheme } from 'react-jss'
import { Theme } from '@naturacosmeticos/natds-themes'
import { Collapse, CollapseProps } from '.'
import Button from '../Button'

const meta: Meta<CollapseProps> = {
  title: 'Utilities/Collapse',
  component: Collapse
};

export default meta;
type Story = StoryObj<CollapseProps>;

export const Playground: Story = {
  render: (args) => {
    const theme: Theme = useTheme()
    const [opened, setOpened] = useState(false)
    const text = opened ? 'hide content' : 'show content'

    return (
      <>
        <Button onClick={() => setOpened(!opened)}>{text}</Button>
        <Collapse {...args} open={opened}>
          <p style={{
            color: theme.color.onSurface,
            fontFamily: theme.typography.fontFamily.primary
          }}
          >
            some content
          </p>
        </Collapse>
      </>
    )
  }
};
