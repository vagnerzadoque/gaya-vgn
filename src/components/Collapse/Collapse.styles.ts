import { createUseStyles } from 'react-jss'
import type { CollapseProps } from './Collapse.props'

const styles = createUseStyles({
  collapse: ({ open = false }: CollapseProps) => ({
    overflow: 'hidden',
    height: open ? 'auto' : 0,
    /**
     * NOTE: this is a trick for the transition to be visible
     * because if we use height as transition type, we would also
     * need to set height manually, instead we use max-height
     */
    maxHeight: open ? '100vh' : 0,
    transition: 'max-height 0.3s ease-in-out'
  })
})

export default styles
