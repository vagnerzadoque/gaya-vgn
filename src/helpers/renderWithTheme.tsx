import React from 'react'
import { JssProvider, SheetsRegistry } from 'react-jss'
import themes from '@naturacosmeticos/natds-themes'
import { render } from '@testing-library/react'
import { ThemeProvider } from '../ThemeProvider'

const renderWithTheme = (component: JSX.Element) => {
  const sheets = new SheetsRegistry()

  const renderTree = render(
    <JssProvider registry={sheets}>
      <ThemeProvider theme={themes.natura.light}>
        {component}
      </ThemeProvider>
    </JssProvider>
  )

  return { styles: sheets.toString(), component: renderTree }
}

export default renderWithTheme
