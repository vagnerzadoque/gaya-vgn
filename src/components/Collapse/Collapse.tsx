import React from 'react'
import styles from './Collapse.styles'
import { CollapseProps } from './Collapse.props'

const Collapse = React.forwardRef<HTMLDivElement, CollapseProps>(({
  className = '',
  testID,
  ...props
}, ref) => {
  const { collapse } = styles(props)

  return (
    <div
      {...props}
      className={`${className} ${collapse}`}
      data-testid={testID}
      ref={ref}
    />
  )
})

export default Collapse
