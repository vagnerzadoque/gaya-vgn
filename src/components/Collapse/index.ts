import Collapse from './Collapse'

export { Collapse }
export { CollapseProps } from './Collapse.props'
export default Collapse
