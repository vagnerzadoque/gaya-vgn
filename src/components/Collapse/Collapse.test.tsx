import React from 'react'
import Collapse from '.'
import renderWithTheme from '../../helpers/renderWithTheme'

describe('Collapse component', () => {
  it('should render correctly with default props and opened', () => {
    const { styles, component } = renderWithTheme(<Collapse open>something</Collapse>)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })

  it('should render correctly with default props and closed', () => {
    const { styles, component } = renderWithTheme(<Collapse open={false}>something</Collapse>)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
})
