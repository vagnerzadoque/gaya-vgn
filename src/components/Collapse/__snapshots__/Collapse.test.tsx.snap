// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse component should render correctly with default props and closed 1`] = `
[
  ".collapse-0-2-1 {}
.collapse-d0-0-2-2 {
  height: 0;
  overflow: hidden;
  max-height: 0;
  transition: max-height 0.3s ease-in-out;
}",
  <div>
    <div
      class=" collapse-0-2-1 collapse-d0-0-2-2"
    >
      something
    </div>
  </div>,
]
`;

exports[`Collapse component should render correctly with default props and opened 1`] = `
[
  ".collapse-0-2-1 {}
.collapse-d0-0-2-2 {
  height: auto;
  overflow: hidden;
  max-height: 100vh;
  transition: max-height 0.3s ease-in-out;
}",
  <div>
    <div
      class=" collapse-0-2-1 collapse-d0-0-2-2"
      open=""
    >
      something
    </div>
  </div>,
]
`;
